net.solarnetwork.node.weather.ibm.wc/src/net/solarnetwork/node/weather/ibm/wc/WCDayDatumDataSource.java

```diff
@@ -0,0 +1,57 @@
+
+package net.solarnetwork.node.weather.ibm.wc;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import net.solarnetwork.node.domain.GeneralDayDatum;
+import net.solarnetwork.node.settings.SettingSpecifier;
+import net.solarnetwork.node.settings.support.BasicTextFieldSettingSpecifier;
+
+public class WCDayDatumDataSource extends WCSupport<GeneralDayDatum> {
+
+	@Override
+	public List<SettingSpecifier> getSettingSpecifiers() {
+		List<SettingSpecifier> result = new ArrayList<SettingSpecifier>(1);
+		result.add(new BasicTextFieldSettingSpecifier("uid", null));
+		result.add(new BasicTextFieldSettingSpecifier("apiKey", null));
+		result.add(new BasicTextFieldSettingSpecifier("datumPeriod", "7day"));
```

Comment:
If these datum periods are actually well-defined and seem quite stable, then a drop-down menu setting would be more appropriate here to make it easier on users. See [this example](https://github.com/SolarNetwork/solarnetwork-node/blob/20bd880e693851ebcc7437758d081e2adc2426a7/net.solarnetwork.node.settings.playpen/src/net/solarnetwork/node/settings/playpen/SettingsPlaypen.java#L130-L138).